sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/epkversion/PartitionProcessorImpl.java

```diff
@@ -247,8 +261,103 @@ public Mono<Void> run(CancellationToken cancellationToken) {
                                         return !cancellationToken.isCancellationRequested() && currentTime.isBefore(stopTimer);
                                     }).flatMap(values -> Flux.empty());
                             }
+                            break;
                         }
-                        break;
+                        case STREAMS_CONSTRAINED: {
+
+                            if (!Configs.isChangeFeedProcessorMalformedResponseRecoveryEnabled()) {
+                                logger.error(
+                                    "Lease with token {}: Encountered streams constrained error and the feature flag to retry on streams constrained is disabled, failing.",
+                                    this.lease.getLeaseToken());
+                                logger.error("Streams constrained exception.", clientException);
+                                this.resultException = new RuntimeException(clientException);
+                                return Flux.error(throwable);
+                            }
+
+                            int retryCount = this.streamsConstrainedRetries.incrementAndGet();
+                            boolean shouldRetry = retryCount <= this.maxStreamsConstrainedRetries;
+
+                            if (!shouldRetry) {
+                                logger.error(
+                                    "Lease with token {}: Reached max retries for streams constrained exception with statusCode [{}] : subStatusCode [{}] : message [{}].",
+                                    this.lease.getLeaseToken(),
+                                    clientException.getStatusCode(),
+                                    clientException.getSubStatusCode(),
+                                    clientException.getMessage());
+                                logger.error("Streams constrained retries exhausted.", clientException);
+                                this.resultException = new RuntimeException(clientException);
+                                return Flux.error(throwable);
+                            }
+
+                            logger.warn(
+                                "Lease with token {}: Streams constrained exception encountered, will retry.",
```

Comment:
we can log the exception in this log statement itself to coorelated the exception with the partition / lease token.

Reply:
Fixed